src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -3163,9 +3163,8 @@ private static bool IsValidExpressionBody(SyntaxNode expressionSyntax, BoundExpr
             Binder bodyBinder = this.GetBinder(expressionBody);
             Debug.Assert(bodyBinder != null);
 
-            RefKind refKind = RefKind.None;
-            ExpressionSyntax expressionSyntax = expressionBody.Expression.CheckAndUnwrapRefExpression(diagnostics, out refKind);
-            BindValueKind requiredValueKind = GetRequiredReturnValueKind(refKind);
+            ExpressionSyntax expressionSyntax = expressionBody.Expression.CheckAndUnwrapRefExpression(diagnostics, out var refKind);
+            BindValueKind requiredValueKind = bodyBinder.GetRequiredReturnValueKind(refKind);
             BoundExpression expression = bodyBinder.BindValue(expressionSyntax, diagnostics, requiredValueKind);
             expression = ValidateEscape(expression, Binder.ExternalScope, refKind != RefKind.None, diagnostics);
```

Comment:
>ValidateEscape [](start = 25, length = 14)

It looks like this is an instance method. Let's use ```bodyBinder``` to call it as well. Since it is very easy to make a mistake and call something on ```this```, consider extracting a helper method, either a static local function, or a new instance method which will be called with only the right binder available to them.